src/power_grid_model/validation/utils.py

```diff
@@ -103,7 +103,22 @@ def update_input_data(input_data: SingleDataset, update_data: SingleDataset):
     return merged_data
 
 
-def update_component_data(component: ComponentTypeLike, input_data: np.ndarray, update_data: np.ndarray) -> None:
+def update_component_data(
+    component: ComponentTypeLike, input_data: SingleComponentData, update_data: SingleComponentData
+) -> None:
+    """
+    Update the data in a single component data set, with another single component data set,
+    indexed on the "id" field and only non-NaN values are overwritten.
+    """
+    if isinstance(input_data, np.ndarray) and isinstance(update_data, np.ndarray):
+        return _update_component_array_data(component=component, input_data=input_data, update_data=update_data)
+
+    raise NotImplementedError()  # TODO(mgovers): add support for columnar data
```

Comment:
@figueroa1395 is this still necessary?

Reply:
It shouldn't be, as all the validation is done in row-based format. As long as we can slice columnar batch data into single scenarios of columnar, this is not needed; and this is only need for the improvement in the batch validation.

Reply:
idem: to be removed in follow-up PR

Reply:
Can you guys agree on this point and resolve the thread (and below)? @mgovers @figueroa1395

Reply:
I'm double checking at the moment. I'll get back in a few.

Reply:
All this are okay to close, double checked all.